app/vmctl/README.md

```diff
@@ -432,6 +436,79 @@ Found 2 blocks to import. Continue? [Y/n] y
+to VictoriaMetrics time-series database.
+
+See `./vmctl remote-read --help` for details and full list of flags.
+
+Migration is based on the remote reading and allows transparently fetch samples. 
+Remote read protocol supports two different modes `SAMPLES` and `STREAMED_XOR_CHUNKS`. You can read more about it 
+[here](https://prometheus.io/blog/2019/10/10/remote-read-meets-streaming/).
+
+Please notice at that moment not all TSDB support `STREAMED_XOR_CHUNKS` mode and can handle stream requests.
+
+To start the migration process some flags should be defined:
+1. `--remote-read-src-addr` - address to perform read from;
+2. `--vm-addr` - address for single-node VM is usually equal to `--httpListenAddr`, and for cluster version;
+is equal to `--httpListenAddr` flag of vminsert component (for example `http://<vminsert>:8480/insert/<accountID>/prometheus`);
+3. `--remote-read-filter-time-start` - the time filter in RFC3339 format to select timeseries with timestamp equal or higher than provided value. E.g. '2020-01-01T20:07:00Z';
+4. `--remote-read-step-interval` - split export data into chunks. Valid values are `month, day, hour, minute`;
+
+As soon as required flags are provided and all endpoints are accessible, `vmctl` will start the fetch process 
+from remote read API. Basically, it just fetches all timeseries from the provided database and sends import requests 
+for each timeseries and pass results to VM importer. VM importer then accumulates received samples in batches and 
+sends import requests to VM.
+
+The importing process example for local installation of Prometheus
+and single-node VictoriaMetrics(`http://localhost:8428`):
+
+```
+./vmctl remote-read \
+--remote-read-src-addr=http://127.0.0.1:9091 \
+--remote-read-filter-time-start=2021-10-18T00:00:00Z \
+--remote-read-step-interval=hour \
+--vm-addr=http://127.0.0.1:8428 \
+--vm-concurrency=6
+
+Split defined times into 8798 ranges to import. Continue? [Y/n]
+VM worker 0:↘ 127177 samples/s
+VM worker 1:↘ 140137 samples/s
+VM worker 2:↘ 151606 samples/s
+VM worker 3:↘ 130765 samples/s
+VM worker 4:↘ 131904 samples/s
+VM worker 5:↘ 132693 samples/s
+Processing ranges: 8798 / 8798 [█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████] 100.00%
+2022/10/19 16:45:37 Import finished!
+2022/10/19 16:45:37 VictoriaMetrics importer stats:
+  idle duration: 6m57.793987511s;
+  time spent while importing: 1m18.463744801s;
+  total samples: 25348208;
+  samples/s: 323056.31;
+  total bytes: 669.7 MB;
+  bytes/s: 8.5 MB;
+  import requests: 127;
+  import requests retries: 0;
+2022/10/19 16:45:37 Total time: 1m19.406283424s
+```
+
+### Filtering
+
+The filtering consists of two parts: by label and label value, and time.
+
+Filtering by time may be configured via flags `--remote-read-filter-time-start` and `--remote-read-filter-time-end`
```

Comment:
Is this true? I thought remote-read API does that, doesn't it?

Reply:
For getting data we should define the next struct
```golang
prompb.Query{
		StartTimestampMs: filter.StartTimestampMs,
		EndTimestampMs:   filter.EndTimestampMs - 1,
		Matchers:         []*prompb.LabelMatcher{m},
}
```

Where we must define the start and the end of collecting data

Reply:
Oh, the comment is mis placed. It was related to the "This filter is applied twice: to drop blocks out of range and to filter timeseries in blocks with overlapping time ranges." part.

Reply:
removed this part of the code. you are right